system7-tests/mergeDriverTests.m

```diff
@@ -641,6 +641,136 @@ - (void)testBothSideUpdateSubrepoToDifferentRevisionConflictResolution_KeepConfl
+
+        const int pushExitStatus = s7push_currentBranch(repo);
+        XCTAssertNotEqual(0, pushExitStatus, @"nik has pushed. I must merge");
+
+        [repo fetch];
+
+        S7ConfigMergeDriver *configMergeDriver = [S7ConfigMergeDriver new];
+
+        [configMergeDriver setResolveConflictBlock:^S7ConflictResolutionOption(S7SubrepoDescription * _Nonnull ourVersion,
+                                                                          S7SubrepoDescription * _Nonnull theirVersion)
+         {
+            XCTAssertEqualObjects(ourVersion.path, @"Dependencies/ReaddleLib");
+
+            return S7ConflictResolutionOptionKeepConflict;
+         }];
+
+        const int mergeExitStatus = [configMergeDriver
+                                     mergeRepo:repo
+                                     baseConfig:baseConfig
+                                     ourConfig:ourConfig
+                                     theirConfig:niksConfig
+                                     saveResultToFilePath:S7ConfigFileName];
+        XCTAssertNotEqual(0, mergeExitStatus);
+
+        S7Config *actualConfig = [[S7Config alloc] initWithContentsOfFile:S7ConfigFileName];
+
+        S7Config *expectedConfig = [[S7Config alloc] initWithSubrepoDescriptions:@[
+            [[S7SubrepoDescriptionConflict alloc]
+             initWithOurVersion:[[S7SubrepoDescription alloc]
+                                 initWithPath:@"Dependencies/ReaddleLib"
+                                 url:self.env.githubReaddleLibRepo.absolutePath
+                                 revision:readdleLib_pasteys_Revision
+                                 branch:@"main"]
+             theirVersion:[[S7SubrepoDescription alloc]
+                           initWithPath:@"Dependencies/ReaddleLib"
+                           url:self.env.githubReaddleLibRepo.absolutePath
+                           revision:readdleLib_niks_Revision
+                           branch:@"main"]],
+
+            [[S7SubrepoDescription alloc]
+             initWithPath:@"Dependencies/RDPDFKit"
+             url:self.env.githubRDPDFKitRepo.absolutePath
+             revision:rdpdfKitLib_niks_Revision
+             branch:@"main"]
+        ]];
+
+        XCTAssertEqualObjects(actualConfig, expectedConfig);
+
+        S7Config *controlConfig = [[S7Config alloc] initWithContentsOfFile:S7ControlFileName];
+        XCTAssertNotNil(controlConfig);
+        XCTAssertEqualObjects(actualConfig, controlConfig);
+
+        // Test ReaddleLib has merge conflict and subrepo is set to local revision.
+        NSString *readdleLibSubrepoRevision = nil;
+        XCTAssertEqual(0, [readdleLibSubrepoGit getCurrentRevision:&readdleLibSubrepoRevision]);
+        XCTAssertEqualObjects(readdleLibSubrepoRevision, readdleLib_pasteys_Revision);
+
+        // Test RDPDFKit has been checked out and points to 'their' revision.
+        GitRepository *pdfkitSubrepoGit = [GitRepository repoAtPath:@"Dependencies/RDPDFKit"];
```

Comment:
Please, check if the ReaddleLib is actually in the conflict state (maybe add a utility function to the `GitRepository` class)